src/rust/engine/process_execution/remote/src/remote_cache.rs

```diff
@@ -65,6 +74,44 @@ pub struct RemoteCacheProviderOptions {
   pub rpc_timeout: Duration,
 }
 
+async fn choose_provider(
+  options: RemoteCacheProviderOptions,
+) -> Result<Arc<dyn ActionCacheProvider>, String> {
+  let address = options.action_cache_address.clone();
+
+  // TODO: we shouldn't need to gin up a whole copy of this struct; it'd be better to have the two
+  // set of remoting options managed together.
```

Comment:
There is a lot of potential cleanup that can happen here, yea.

Ideally we could expose a struct all the way through to Python (with a very light Python wrapper around the inner type), such that it could directly be constructed from global options. Currently that's not the case -- in particular: see [ExecutionOptions](https://github.com/pantsbuild/pants/blob/8097749a3268b5f3e302f388a8df327539de1904/src/python/pants/option/global_options.py#L475), which is copied into positional args to the scheduler.